workflowtrigger/api/pipeline_api.go

```diff
@@ -212,12 +213,63 @@ func deletePipeline(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
+	if err := clearPipelineData(orgName, teamName, pipelineName); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
 
 	dbClient.StoreValue(key, teamSchema)
 	w.WriteHeader(http.StatusOK)
 	return
 }
 
+func clearPipelineData(orgName string, teamName string, pipelineName string) error {
+	prefix := fmt.Sprintf("%s-%s-%s", orgName, teamName, pipelineName)
+	return dbClient.DeleteByPrefix(prefix)
+}
+
+func updatePipeline(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	orgName := vars[orgNameField]
+	teamName := vars[teamNameField]
+	pipelineName := vars[pipelineNameField]
+	key := db.MakeDbTeamKey(orgName, teamName)
+
+	var gitRepoUpd git.GitRepoSchema
+	buf := new(bytes.Buffer)
+	_, err := buf.ReadFrom(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	gitRepoUpd = git.UnmarshallGitRepoSchemaString(string(buf.Bytes()))
+
+	teamSchema := dbClient.FetchTeamSchema(key)
+	if teamSchema.GetOrgName() == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if teamSchema.GetPipelineSchema(pipelineName) == nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if !schemaValidator.ValidateSchemaAccess(orgName, teamName, teamSchema.GetPipelineSchema(pipelineName).GetGitRepoSchema().GitRepo, reposerver.RootCommit, string(argoauthenticator.UpdateAction), string(argoauthenticator.ApplicationResource)) {
+		http.Error(w, "Not enough permissions", http.StatusForbidden)
+	}
+	kubernetesClient.StoreGitCred(nil, db.MakeSecretName(orgName, teamName, pipelineName))
```

Comment:
The new gitRepo (gitRepoUpd) should be passed to StoreGitCred (instead of nil) to update the credentials in K8S.

Reply:
Passed git cred value from gitRepoUpd